### DISCUSSION TITLE
AWS IPC Pub Sub vs Moquette

### DISCUSSION THREAD

**Discussion by chandar1** [aws/aws-iot-device-sdk-cpp-v2#760] [Q&A] [answered]
Hi,

We have a need for inter communication between Green Grass components running in a target device. Evaluating options to use IPC pub/Sub for local/inter process communication and/or using Moquette broker running in target device for the same inter process communication. 

Are there any bench mark metrics on performance or the time it takes for messages to be sent/received. 

Plus, IPC Pub/Sub deviates from MQTT standard - it doesn't have QoS option or Retain flag. What are the reasons for this? 
We would want the internal broker to retain our messages, what can be done for it.

Your answers are appreciated.

**Comment by bretambrose** [comment, marked as answer]
We don't have any benchmarks on IPC performance.

For the other questions, a better target would be https://github.com/aws-greengrass/aws-greengrass-nucleus

FWIW, my understanding is that the greengrass nucleus embeds a moquette broker internally and the local ipc pub/sub APIs are accessing it directly.
